### PR TITLE
Add new pass for math to rocdl.

### PR DESCRIPTION
**Pull request by jsjodin**
There are math operations that are not yet being converted, but it works for most operations. The changes to the flang codegen enables the pass for AMDGCN and disables the MathToLLVM and MathToLibm passes.

### CODE REVIEW

File: flang/lib/Optimizer/CodeGen/CodeGen.cpp
Comment (agozillon) [inline flang/lib/Optimizer/CodeGen/CodeGen.cpp:3564]: Nit: would it be possible to get the full name for ocml in the comments? Maybe a weird request, just unfamiliar with the library so would be handy to have in the comment! if not that's not a big deal.

File: flang/lib/Optimizer/CodeGen/CodeGen.cpp
Comment (agozillon) [inline flang/lib/Optimizer/CodeGen/CodeGen.cpp:3579]: Nit: Unnecessary space addition

**Review by agozillon — APPROVED** [review]
The PR LGTM :-) feel free to wait on a secondary reviewer approval or not, I'll leave it up to you!

File: flang/lib/Optimizer/CodeGen/CodeGen.cpp
Comment (jsjodin) [reply flang/lib/Optimizer/CodeGen/CodeGen.cpp:3564]: I think ocml stands for OpenCL Math Library, so I'm not sure that will help or confuse more. I should perhaps use "AMD GPU math library" or something like that.

File: flang/lib/Optimizer/CodeGen/CodeGen.cpp
Comment (skatrak) [inline flang/lib/Optimizer/CodeGen/CodeGen.cpp:3568]: Nit: Braces

**Review by skatrak — APPROVED** [review]
Thank you Jan, LGTM as well. It'd be good to add a Lit test for this as well, to make sure it doesn't inadvertently break later on.

**Pull request merged**
